### ISSUE TITLE
Minimal Implementation with TF2 / `tf.distribute.Strategy`

### ISSUE THREAD

**Issue by tonouchi510** [google-research/simclr#103]
Hi Ting!

I’ve done a minimal implementation of SimCLR that supports distributed learning by tf2.

Since SimCLR is recommended to be trained in huge batch sizes, I thought this implementation might be useful and would like to share it with you.
https://github.com/tonouchi510/simclr-tf2-distribute

The features of this implementation are as follows
- Minimal implementation of SimCLR with `tf2` / `distribute.strategy`
  - Assumed to experiment using cost-effective preemptive-tpu v3-8, used tpu_strategy.
- Simple and extensible implementation with `tf.keras.Model.fit`

Thank you!
Tonouchi

**Comment by tonouchi510** [comment]
May I ask you to add a link to the README? 
since I read you are accepting implementations.

**Comment by chentingpc** [comment]
Hi there, thanks for sharing your implementation. I was wondering if you have some results produced by the repo? Also it seems you didn't have global/synchronized BN which may lead to issues when training with multiple GPUs/TPUs.

**Comment by tonouchi510** [comment]
Sorry for the late reply.
My implementation was based on [sayakpaul/SimCLR-in-TensorFlow-2](https://github.com/sayakpaul/SimCLR-in-TensorFlow-2), so I used a simple model structure as well.

I'm still in the process of experimenting and improving it. 
As a future work, I would like to conduct experiments to compare with and without BN, referring to the official implementation as well.

As for the experimental results, we are not able to post them because we are currently using business data (face image), but from the learning curve and Linear Evaluation results, we have confirmed that the training is proceeding without problems.
This is also a feature work, but I will post the results of the experiment with public data when I have more time.
